RockWeb/Blocks/Examples/ModelMap.ascx.cs

```diff
@@ -56,15 +84,138 @@ protected override void OnLoad( EventArgs e )
+            }
+        }
+
+        /// <summary>
+        /// Handles the ItemCommand event of the rptCategory control.
+        /// </summary>
+        /// <param name="source">The source of the event.</param>
+        /// <param name="e">The <see cref="RepeaterCommandEventArgs"/> instance containing the event data.</param>
+        protected void rptCategory_ItemCommand( object source, RepeaterCommandEventArgs e )
+        {
+            string category = e.CommandArgument.ToString();
+            SelectedCategory = category;
+            SelectedModelId=null;
+            GetData();
+        }
+
+        /// <summary>
+        /// Handles the ItemCreated event of the rptModel control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="RepeaterItemEventArgs"/> instance containing the event data.</param>
+        protected void rptModel_ItemCreated( object sender, RepeaterItemEventArgs e )
+        {
+            var lbModel = e.Item.FindControl( "lbModel" ) as LinkButton;
+            if ( lbModel != null )
+            {
+                ScriptManager.GetCurrent( this.Page ).RegisterPostBackControl( lbModel );
+            }
+        }
+
+        /// <summary>
+        /// Handles the ItemCommand event of the rptModel control.
+        /// </summary>
+        /// <param name="source">The source of the event.</param>
+        /// <param name="e">The <see cref="RepeaterCommandEventArgs"/> instance containing the event data.</param>
+        protected void rptModel_ItemCommand( object source, RepeaterCommandEventArgs e )
+        {
+            Guid guidModel = e.CommandArgument.ToString().AsGuid();
+            SelectedModelId = guidModel;
+            GetData();
+        }
+
+        #endregion
+
+        #region Methods
+
+        private void GetData()
+        {
             var sb = new StringBuilder();
 
             List<MClass> allClasses = GetModelClasses() as List<MClass>;
             if ( allClasses != null )
-            { 
-                foreach ( var aClass in allClasses.OrderBy( a => a.Name ).ToList() )
+            {
+                var qry = allClasses.GroupBy( a => a.Category.Name )
+                    .OrderBy( a => a.Key );
+
+                if ( string.IsNullOrWhiteSpace( SelectedCategory ) )
                 {
-                    sb.Append( ClassNode( aClass ) );
+                    SelectedCategory = qry.Select( a => a.Key ).First();
                 }
+
+                var displayCategoryList = qry.
+                                    Select( a => new
+                                    {
+                                        Category = new Category
+                                        {
+                                            Name = a.Key,
+                                            IconCssClass = a.Select( g => g.Name ).First(),
```

Comment:
Not sure this is actually getting the IconCssClass. Shouldn't it be something like 
```
IconCssClass = a.Select( g=> g.Category.IconCssClass ).FirstOrDefault()
```